src/Framework/AspNetCoreAnalyzers/src/CodeFixes/Authorization/AddAuthorizationBuilderFixer.cs

```diff
@@ -0,0 +1,135 @@
+            && arguments[0].Expression is SimpleLambdaExpressionSyntax lambda)
+        {
+            IEnumerable<SyntaxNode> nodes;
+
+            if (lambda.Body is BlockSyntax lambdaBlockBody)
+            {
+                nodes = lambdaBlockBody.DescendantNodes();
+            }
+            else if (lambda.Body is InvocationExpressionSyntax lambdaExpressionBody)
+            {
+                nodes = new[] { lambdaExpressionBody };
+            }
+            else
+            {
+                return false;
+            }
+
+            var addAuthorizationBuilderMethod = memberAccessExpression.ReplaceToken(identifierToken,
+                SyntaxFactory.Identifier("AddAuthorizationBuilder"));
+
+            invocation = SyntaxFactory.InvocationExpression(addAuthorizationBuilderMethod);
+
+            foreach (var configureAction in nodes)
+            {
+                if (configureAction is InvocationExpressionSyntax { ArgumentList.Arguments: { Count: 2 } configureArguments, Expression: MemberAccessExpressionSyntax { Name.Identifier.Text: "AddPolicy" } })
+                {
+                    invocation = ChainInvocation(
+                        invocation,
+                        "AddPolicy",
+                        SyntaxFactory.ArgumentList(
+                            SyntaxFactory.SeparatedList(configureArguments)));
+                }
+                else if (configureAction is AssignmentExpressionSyntax { Left: MemberAccessExpressionSyntax { Name.Identifier.Text: { } assignmentTargetName }, Right: { } assignmentExpression }
+                    && assignmentTargetName is "DefaultPolicy" or "FallbackPolicy" or "InvokeHandlersAfterFailure")
+                {
+                    invocation = ChainInvocation(
+                        invocation,
+                        $"Set{assignmentTargetName}",
+                        SyntaxFactory.ArgumentList(
+                            SyntaxFactory.SingletonSeparatedList(
+                                SyntaxFactory.Argument(assignmentExpression))));
+                }
+            }
+
+            return true;
+        }
+
+        return false;
+    }
+
+    private static InvocationExpressionSyntax ChainInvocation(
+        InvocationExpressionSyntax invocation,
+        string invokedMemberName,
+        ArgumentListSyntax argumentList)
+    {
+        return SyntaxFactory.InvocationExpression(
+            SyntaxFactory.MemberAccessExpression(
+                SyntaxKind.SimpleMemberAccessExpression,
+                invocation.WithTrailingTrivia(
+                    SyntaxFactory.EndOfLine(Environment.NewLine),
+                    SyntaxFactory.Space,
+                    SyntaxFactory.Space,
+                    SyntaxFactory.Space,
+                    SyntaxFactory.Space),
```

Comment:
This part seems to be the sketchiest to me. I guess our tests expect `SyntaxFactory.EndOfLine("\r\n")`. Is this indentation relative to the previous line? Can we add tests for a more deeply nested call into `AddAuthorization(...)` like in `HostBuilder.ConfigureServices`?

Reply:
I agree. This part still felt pretty hacky to me, which is why I left the PR in draft. I haven't confirmed it yet, but I'm assuming this may also related to the test that's failing due to some line ending issues.

I added this in an attempt to improve the formatting of the code generated by the code fix so that it would match the example "after" code example in the issue, otherwise there wouldn't be any indentation relative to the previous line. 

Should we be concerned with the fixed code being "pretty" (within reason) or should we leave this up to the user to handle? If we do want to handle formatting as part of the code fix, I can go ahead and add tests for more deeply nested scenarios. Otherwise, I'll remove this and update the `fixedSource` values in the tests.

Fixed example: without the added indentation and newline
```csharp
builder.Services.AddAuthorizationBuilder().AddPolicy("AtLeast18", policy =>
        policy.Requirements.Add(new MinimumAgeRequirement(18))).AddPolicy("AtLeast21", policy =>
        policy.Requirements.Add(new MinimumAgeRequirement(21)));
```

Reply:
I think we want it to look reasonably pretty if we're going to do this. The example without the added indentation and newline does look bad.

Reply:
How helpful is the `SyntaxNode.NormalizeWhitespace` API here for formatting the `InvocationExpression`? I know it sometimes doesn't play too nicely with deeply-nested calls but wondering if it might help here.

Reply:
Following up on this. Calling `NormalizeWhitespace` on the final `InvocationExpression` actually seems to format the block body examples OK, although the expression body examples ended up being pretty long since `NormalizeWhitespace` doesn't add new lines after the chained calls. So it looks like we'll need to add the `EndOfLineTrivia` ourselves if we want go this route.

Example: `MultipleAddPolicyCalls_UsingBlockBody_FixedWithAddAuthorizationBuilder`
```csharp
builder.Services.AddAuthorizationBuilder().AddPolicy("AtLeast18", policy =>
{
    policy.Requirements.Add(new MinimumAgeRequirement(18));
}).AddPolicy("AtLeast21", policy =>
{
    policy.Requirements.Add(new MinimumAgeRequirement(21));
});
```

Example: `MultipleAddPolicyCalls_UsingExpressionBody_FixedWithAddAuthorizationBuilder`
```csharp
builder.Services.AddAuthorizationBuilder().AddPolicy("AtLeast18", policy => policy.Requirements.Add(new MinimumAgeRequirement(18))).AddPolicy("AtLeast21", policy => policy.Requirements.Add(new MinimumAgeRequirement(21)));
```

Reply:
I also added two tests for the nested `AddAuthorization` scenarios and made some adjustments to have leading trivia is handled in `AddAuthorizationBuilderFixer` to ensure that indentation is correct.